src/tests/import.rs

```diff
@@ -1934,3 +1934,76 @@ fn import_criteria_map_aggregated_error() {
 
     insta::assert_snapshot!(output);
 }
+
+#[test]
+fn import_subsumed_by_local_wildcard_audit() {
+    // (Pass) If a local wildcard audit accounts for a crate, an imported audit should be pruned
+    // (the local wildcard audit is preferred).
+
+    let _enter = TEST_RUNTIME.enter();
+    let mock = MockMetadata::simple();
+
+    let metadata = mock.metadata();
+    let (mut config, mut audits, mut imports) = builtin_files_full_audited(&metadata);
+
+    audits.audits.remove("third-party2");
+
+    audits.wildcard_audits.insert(
+        "third-party2".to_owned(),
+        vec![wildcard_audit(1, SAFE_TO_DEPLOY)],
+    );
+
+    let foreign_audits = AuditsFile {
+        criteria: SortedMap::new(),
+        audits: [(
+            "third-party2".to_owned(),
+            vec![full_audit(ver(DEFAULT_VER), SAFE_TO_DEPLOY)],
+        )]
+        .into(),
+        wildcard_audits: [(
+            "third-party2".to_owned(),
+            vec![wildcard_audit(1, SAFE_TO_DEPLOY)],
+        )]
+        .into(),
+        trusted: SortedMap::new(),
+    };
+
+    imports
+        .audits
+        .insert(FOREIGN.to_owned(), foreign_audits.clone());
+
+    imports.publisher.insert(
+        "third-party2".to_owned(),
+        vec![publisher_entry_named(
+            ver(DEFAULT_VER),
+            1,
+            "user1",
+            "User One",
+        )],
+    );
```

Comment:
This publisher entry shouldn't be pre-populated within `imports.lock` - we should be fetching it from the network to model the situation better.

Reply:
Oh right, so I was testing in a situation where we did actually already have it there (import first, then add a wildcard audit). I can change this or make a separate test to test fresh imports.